src/io/flutter/actions/OpenInAndroidStudioAction.java

```diff
@@ -30,11 +30,14 @@
 
 public class OpenInAndroidStudioAction extends AnAction {
 
-  private static void openFileInStudio(@NotNull VirtualFile file, @NotNull String androidStudioPath) {
+  private static void openFileInStudio(@NotNull VirtualFile file, @NotNull String androidStudioPath, @Nullable String sourceFile) {
     try {
       final GeneralCommandLine cmd;
       if (SystemInfo.isMac) {
-        cmd = new GeneralCommandLine().withExePath("open").withParameters("-a", androidStudioPath, file.getPath());
+        cmd = new GeneralCommandLine().withExePath("open").withParameters("-a", androidStudioPath, "--args", file.getPath());
+        if (sourceFile != null) {
+          cmd.addParameter(sourceFile);
+        }
       }
       else {
         cmd = new GeneralCommandLine().withExePath(androidStudioPath).withParameters(file.getPath());
```

Comment:
Is this known not to work for Windows or is it just unexplored?  (If it's the latter, maybe a TODO?)

Reply:
It is known to work for Mac. I added a TODO.